src/storage/types.rs

```diff
@@ -106,7 +106,7 @@ pub struct Fingerprint {
     pub card_fingerprint: Secret<String>,
 }
 
-#[derive(Debug, Clone, Serialize)]
+#[derive(Debug, Clone, Serialize, serde::Deserialize)]
```

Comment:
There is a API struct that can be used the from `data/types.rs` but instead, we can use the CardNumber struct